### PR TITLE
feat: Getting production calculations as a grid operator

### PR DESCRIPTION
**Pull request by RasmusGraabaek**
## Description

We want to extract calculations for the time series type: production as a grid operator.

## References

* #0000
This is part of the ongoing story:
https://app.zenhub.com/workspaces/mosaic-60a6105157304f00119be86e/issues/gh/energinet-datahub/team-phoenix/60

### CODE REVIEW

File: source/dotnet/wholesale-api/Events/Events.Infrastructure/InboxEvents/AggregatedTimeSeriesMessageFactory.cs [hunk 1]
```diff
@@ -14,10 +14,13 @@
 
 using Azure.Messaging.ServiceBus;
 using Energinet.DataHub.Edi.Responses;
+using Energinet.DataHub.Wholesale.CalculationResults.Interfaces.CalculationResults.Model;
 using Energinet.DataHub.Wholesale.Events.Application.InboxEvents;
 using Google.Protobuf;
 using Google.Protobuf.WellKnownTypes;
 using PeriodContract = Energinet.DataHub.Edi.Responses.Period;
+using QuantityQuality = Energinet.DataHub.Edi.Responses.QuantityQuality;
+using TimeSeriesPoint = Energinet.DataHub.Edi.Responses.TimeSeriesPoint;
```
Comment (RasmusGraabaek) [inline source/dotnet/wholesale-api/Events/Events.Infrastructure/InboxEvents/AggregatedTimeSeriesMessageFactory.cs:23]:
```suggestion
```

File: source/dotnet/wholesale-api/Events/Events.Infrastructure/InboxEvents/AggregatedTimeSeriesMessageFactory.cs [hunk 2]
```diff
@@ -14,10 +14,13 @@
...
 using TimeSeriesTypeContract = Energinet.DataHub.Edi.Responses.TimeSeriesType;
```
Comment (RasmusGraabaek) [inline source/dotnet/wholesale-api/Events/Events.Infrastructure/InboxEvents/AggregatedTimeSeriesMessageFactory.cs:24]:
```suggestion
using PeriodContract = Energinet.DataHub.Edi.Responses.Period;
using TimeSeriesTypeContract = Energinet.DataHub.Edi.Responses.TimeSeriesType;
```

File: source/dotnet/wholesale-api/Events/Events.Infrastructure/InboxEvents/AggregatedTimeSeriesMessageFactory.cs (hunk 1 above)
Comment (RasmusGraabaek) [inline source/dotnet/wholesale-api/Events/Events.Infrastructure/InboxEvents/AggregatedTimeSeriesMessageFactory.cs:23]:
```suggestion
```

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Infrastructure/CalculationResults/CalculationResultQueriesTests.cs [hunk 3]
```diff
@@ -76,6 +78,71 @@ public CalculationResultQueriesTests(DatabricksSqlStatementApiFixture fixture)
             .Equal(FirstQuantity, SecondQuantity, ThirdQuantity, FourthQuantity, FifthQuantity, SixthQuantity);
     }
 
+    [Theory]
+    [InlineAutoMoqData]
+    public async Task GetAsync_RequestFromTotalProduction_ReturnsMultipleResults(
```
Comment (RasmusGraabaek) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Infrastructure/CalculationResults/CalculationResultQueriesTests.cs:83]:
```suggestion
    public async Task GetAsync_RequestTotalProduction_ReturnSingleResult(
```

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResults/CalculationResultQueries.cs [hunk 4]
```diff
@@ -135,4 +188,31 @@ private static TimeSeriesPoint CreateTimeSeriesPoint(SqlResultRow row)
...
+        var batchId = sqlResultRow[EnergyResultColumnNames.BatchId];
+        return new CalculationResult(
+            id,
+            Guid.Parse(batchId),
+            gridArea,
+            timeSeriesType,
+            energySupplierId,
+            balanceResponsibleId,
+            timeSeriesPoints.ToArray(),
+            // Should we load this from the batch?
```
Comment (lasrinnil) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResults/CalculationResultQueries.cs:212]: I think this should be a part of the request from EDI?

File: source/dotnet/wholesale-api/Events/Events.Application/InboxEvents/AggregatedTimeSeriesRequest.cs [hunk 5]
```diff
@@ -15,6 +15,7 @@
 namespace Energinet.DataHub.Wholesale.Events.Application.InboxEvents;
 
 public record AggregatedTimeSeriesRequest(
+    AggregationLevel AggregationLevel,
```
Comment (lasrinnil) [inline source/dotnet/wholesale-api/Events/Events.Application/InboxEvents/AggregatedTimeSeriesRequest.cs:18]: There are too many identifiers for the aggregationlevel?

File: source/dotnet/wholesale-api/Events/Events.Application/InboxEvents/IAggregatedTimeSeriesMessageFactory.cs [hunk 6]
```diff
@@ -21,8 +22,8 @@ public interface IAggregatedTimeSeriesMessageFactory
     /// <summary>
     /// Creates a service bus message based on aggregated time series
     /// </summary>
-    /// <param name="aggregatedTimeSeries"></param>
+    /// <param name="calculationResults"></param>
```
Comment (lasrinnil) [inline source/dotnet/wholesale-api/Events/Events.Application/InboxEvents/IAggregatedTimeSeriesMessageFactory.cs:25]: remove paras or describe them imo :)

File: source/dotnet/wholesale-api/Events/Events.Application/UseCases/AggregatedTimeSeriesRequestHandler.cs [hunk 7]
```diff
@@ -39,17 +44,54 @@ public class AggregatedTimeSeriesRequestHandler : IAggregatedTimeSeriesRequestHa
...
+            aggregatedTimeSeriesRequestMessage.Period.Start,
+            aggregatedTimeSeriesRequestMessage.Period.End,
+            MapGridAreaCode(aggregatedTimeSeriesRequestMessage),
+            MapAggregationLevel(aggregatedTimeSeriesRequestMessage.AggregationLevel));
+
+        return await _calculationResultQueries.GetAsync(query)
+            .ToListAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
+    }
+
+    private static CalculationAggregationLevel MapAggregationLevel(AggregationLevel aggregationLevel)
```
Comment (lasrinnil) [inline source/dotnet/wholesale-api/Events/Events.Application/UseCases/AggregatedTimeSeriesRequestHandler.cs:76]: Perhaps these "Map" methods should be moved to seperate files and tested, this domain differs from our own testwise.

File: source/dotnet/wholesale-api/Events/Events.Application/Events.Application.csproj [hunk 8]
```diff
@@ -12,6 +12,7 @@
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
+    <PackageReference Include="System.Linq.Async" Version="6.0.1" />
```
Comment (lasrinnil) [inline source/dotnet/wholesale-api/Events/Events.Application/Events.Application.csproj:15]: move to <PacakgeReference> itemgroup above

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Infrastructure/CalculationResults/CalculationResultQueriesTests.cs [hunk 9]
```diff
@@ -76,8 +79,75 @@ public CalculationResultQueriesTests(DatabricksSqlStatementApiFixture fixture)
...
+
+        var request = CreateRequest();
+        var sut = new CalculationResultQueries(sqlStatementClient, batchesClientMock.Object, deltaTableOptions, calculationResultQueriesLoggerMock.Object);
+
+        // Act
+        var actual = await sut.GetAsync(request).ToListAsync();
+
+        // Assert
+        actual.Should().NotBeNull();
+        actual.Count.Should().Be(expectedResultsCount);
```
Comment (lasrinnil) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Infrastructure/CalculationResults/CalculationResultQueriesTests.cs:104]: This test should perhaps inspect that the retrieved points "time" are within the searchcriteria eg:
GridArea, time, AggregationLevel?

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Infrastructure/CalculationResults/CalculationResultQueriesTests.cs [hunk 10]
```diff
@@ -76,8 +79,75 @@ public CalculationResultQueriesTests(DatabricksSqlStatementApiFixture fixture)
...
+            startOfPeriod: Instant.FromUtc(2020, 1, 1, 1, 1),
+            endOfPeriod: Instant.FromUtc(2021, 1, 2, 1, 1));
+        var sut = new CalculationResultQueries(sqlStatementClient, batchesClientMock.Object, deltaTableOptions, calculationResultQueriesLoggerMock.Object);
+
+        // Act
+        var actual = await sut.GetAsync(request).ToListAsync();
+
+        // Assert
+        actual.Should().NotBeNull();
+        actual.Count.Should().Be(expectedResultCount);
```
Comment (lasrinnil) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Infrastructure/CalculationResults/CalculationResultQueriesTests.cs:131]: Should.Be.Empty()?

**Review by johevemi — APPROVED** [review]

**Pull request merged**